tests/CMakeLists.txt

```diff
@@ -6,16 +6,128 @@ ADD_DEFINITIONS(-DTESTS_DATADIR="${CMAKE_CURRENT_SOURCE_DIR}/data")
+#
+#    Create target ctest_build_tests if does not exist yet.
+#
+MACRO(CREATE_CTEST_BUILD_TESTS_TARGET)
+  GET_PROPERTY(ctest_build_tests_exists_value GLOBAL PROPERTY ctest_build_tests_exists)
+  IF(NOT BUILD_TESTING)
+    IF(NOT ctest_build_tests_exists_value)
+      ADD_TEST(ctest_build_tests "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target build_tests -- $ENV{MAKEFLAGS})
+      SET_PROPERTY(GLOBAL PROPERTY ctest_build_tests_exists ON)
+    ENDIF(NOT ctest_build_tests_exists_value)
+  ENDIF(NOT BUILD_TESTING)
+ENDMACRO(CREATE_CTEST_BUILD_TESTS_TARGET)
+
+#.rst:
+# .. command:: ADD_UNIT_TEST (NAME SOURCE)
+#
+#   The behaviour of this function depends on :variable:`BUILD_TESTING` option.
+#
+MACRO(ADD_UNIT_TEST NAME SOURCE)
+  CREATE_CTEST_BUILD_TESTS_TARGET()
+
+  IF(NOT BUILD_TESTING)
+    ADD_EXECUTABLE(${NAME} EXCLUDE_FROM_ALL ${SOURCE})
+  ELSE(NOT BUILD_TESTING)
+    ADD_EXECUTABLE(${NAME} ${SOURCE})
+  ENDIF(NOT BUILD_TESTING)
+
+  ADD_DEPENDENCIES(build_tests ${NAME})
+
+  ADD_TEST(${NAME} ${RUNTIME_OUTPUT_DIRECTORY}/${NAME})
+  # Support definition of DYLD_LIBRARY_PATH for OSX systems
+  IF(APPLE)
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH}")
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH}")
+  ENDIF(APPLE)
+
+  IF(NOT BUILD_TESTING)
+    SET_TESTS_PROPERTIES(${NAME} PROPERTIES DEPENDS ctest_build_tests)
+  ENDIF(NOT BUILD_TESTING)
+ENDMACRO(ADD_UNIT_TEST NAME SOURCE)
+
+#.rst:
+# .. command:: ADD_PYTHON_UNIT_TEST (NAME SOURCE [MODULES...])
+#
+#   Add a test called `NAME` that runs an equivalent of ``python ${SOURCE}``,
+#   optionnaly with a `PYTHONPATH` set to `CMAKE_BINARY_DIR/MODULE_PATH` for each MODULES
+#   `SOURCE` is relative to `PROJECT_SOURCE_DIR`
+#
+#   .. note:: :command:`FINDPYTHON` should have been called first.
+#
+MACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
+  ADD_TEST(NAME ${NAME} COMMAND ${PYTHON_EXECUTABLE} "${PROJECT_SOURCE_DIR}/${SOURCE}")
+  SET(PYTHONPATH)
+
+  SET(MODULES "${ARGN}")  # ARGN is not a variable
+  FOREACH(MODULE_PATH IN LISTS MODULES)
+    LIST(APPEND PYTHONPATH "${CMAKE_BINARY_DIR}/${MODULE_PATH}")
+  ENDFOREACH(MODULE_PATH IN LISTS MODULES)
+
+  IF(DEFINED ENV{PYTHONPATH})
+    LIST(APPEND PYTHONPATH "$ENV{PYTHONPATH}")
+  ENDIF(DEFINED ENV{PYTHONPATH})
+
+  # get path separator to join those paths
+  EXECUTE_PROCESS(COMMAND
+      "${PYTHON_EXECUTABLE}" "-c" "import os; print(os.pathsep)"
+      OUTPUT_VARIABLE PATHSEP
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  STRING(REPLACE ";" "${PATHSEP}" PYTHONPATH_STR "${PYTHONPATH}")
+  SET(ENV_VARIABLES "PYTHONPATH=${PYTHONPATH_STR}")
+  IF(APPLE)
+    LIST(APPEND ENV_VARIABLES "LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH}")
+    LIST(APPEND ENV_VARIABLES "DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH}")
+  ENDIF(APPLE)
+  SET_TESTS_PROPERTIES(${NAME} PROPERTIES ENVIRONMENT "${ENV_VARIABLES}")
+ENDMACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
```

Comment:
This looks like the content of cmake/test.cmake. Any reason to copy it here ?

Reply:
ADD_UNIT_TEST was not found
When i tried to build after the prerelease it couldn't import cmake/test.cmake when i tried INCLUDE(test.cmake)
It was on a docker maybe it is because of that

Reply:
Your docker probably lacked the git submodule. you should always use `git clone --recursive`to run your `git clone`, and if you can't, you have to manually get the submodule with `git submodule update --init`